pkg/query-service/utils/format_test.go

```diff
@@ -0,0 +1,221 @@
+package utils
+
+import (
+	"reflect"
+	"testing"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+type args struct {
+	v        interface{}
+	dataType v3.AttributeKeyDataType
+}
+
+var testValidateAndCastValueData = []struct {
+	name    string
+	args    args
+	want    interface{}
+	wantErr bool
+}{
+	// Test cases for v3.AttributeKeyDataTypeString
+	{
+		name: "v3.AttributeKeyDataTypeString: Valid string",
+		args: args{
+			v:        "test",
+			dataType: v3.AttributeKeyDataTypeString,
+		},
+		want:    "test",
+		wantErr: false,
+	},
+	{
+		name: "v3.AttributeKeyDataTypeString: Valid int",
+		args: args{
+			v:        1,
+			dataType: v3.AttributeKeyDataTypeString,
+		},
+		want:    "1",
+		wantErr: false,
+	},
+	{
+		name: "v3.AttributeKeyDataTypeString: Valid float32",
+		args: args{
+			v:        float32(1.1),
+			dataType: v3.AttributeKeyDataTypeString,
+		},
+		want:    "1.1",
+		wantErr: false,
+	},
+	{
+		name: "v3.AttributeKeyDataTypeString: Valid float64",
+		args: args{
+			v:        float64(1.1),
+			dataType: v3.AttributeKeyDataTypeString,
+		},
+		want:    "1.1",
+		wantErr: false,
+	},
+	{
+		name: "v3.AttributeKeyDataTypeString: Valid bool",
+		args: args{
+			v:        true,
+			dataType: v3.AttributeKeyDataTypeString,
+		},
+		want:    "true",
+		wantErr: false,
+	},
+	{
+		name: "v3.AttributeKeyDataTypeString: Valid []interface{}",
+		args: args{
+			v:        []interface{}{"test"},
+			dataType: v3.AttributeKeyDataTypeString,
+		},
+		want:    []interface{}{"test"},
+		wantErr: false,
+	},
+	{
+		name: "v3.AttributeKeyDataTypeString: Invalid type",
+		args: args{
+			v:        map[string]interface{}{"test": "test"},
```

Comment:
And similarly, what happens if this is a mix of numbers and strings in JSON?

Reply:
We are not supporting `map[string]interface{}{"test": "test"}` so it should return error